conn.go

```diff
@@ -89,6 +89,7 @@ type Config struct {
 	PSKs             PreSharedKeyCache
 	PSKModes         []PSKKeyExchangeMode
 	NonBlocking      bool
+	DatagramMode     bool
```

Comment:
Maybe it's just me, but `DatagramMode` sounds like an enum, like there might be multiple modes.  Maybe just call this option `DTLS` or `UseDTLS`?